### PR TITLE
Replace CompositeMatchAnd subclass with Match#all() (part 2)

### PR DESCRIPTION
**Pull request by ssoloff**
This is one part of multiple PRs whose goal is to remove the `CompositeMatchAnd` subclass and replace its instances with calls to the new `Match#all()` factory method or use of the `Match$CompositeBuilder` class.

There were three types of `CompositeMatchAnd` usage throughout the codebase:

* single-step initialization
* multi-step initialization without branching
* multi-step initialization with branching

This PR addresses multi-step initialization without branching, which is described below along with an example of how it was replaced.

#### Multi-step initialization without branching

This is the case where the `CompositeMatchAnd` constructor is typically called with no arguments and subsequent calls are made to `CompositeMatch#add()`.  However, the code path through the `add()` calls does not branch, and the collection of component matches is known at compile time.  Thus, the code can be replaced by passing all component matches directly to the constructor.  For example, the following code:

```java
final CompositeMatch<Unit> alliedCarrier = new CompositeMatchAnd<>();
alliedCarrier.add(Matches.UnitIsCarrier);
alliedCarrier.add(Matches.alliedUnit(defender, data));
```

would be replaced with:

```java
final Match<Unit> alliedCarrier = Match.all(Matches.UnitIsCarrier, Matches.alliedUnit(defender, data));
```

### CODE REVIEW

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java [hunk 1]
```diff
@@ -480,8 +480,7 @@ private void removeAirNoLongerInTerritory() {
     // remove any air units that were once in this attack, but have now
     // moved out of the territory
     // this is an ilegant way to handle this bug
-    final CompositeMatch<Unit> airNotInTerritory = new CompositeMatchAnd<>();
-    airNotInTerritory.add(Matches.unitIsInTerritory(m_battleSite).invert());
+    final Match<Unit> airNotInTerritory = Matches.unitIsInTerritory(m_battleSite).invert();
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: There was only one component match passed to the `CompositeMatchAnd` constructor here, so I didn't even bother to replace it with a `Match#all()` call.

I'm not sure, but there may be a bug here.  It looks like this removes all units not in the territory, not just air units.  Although, this could have been an intentional change, and the author simply didn't bother to rename the method.  I'll see if I can track it down in the history.

Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: The method `MustFightBattle#removeAirNoLongerInTerritory()` was originally added in b3b6391:

```java
private void removeAirNoLongerInTerritory()
{
    //remove any air units that were once in this attack, but have now moved out of the territory
    //this is an ilegant way to handle this bug
    CompositeMatch airNotInTerritory = new CompositeMatchAnd();
    airNotInTerritory.add( new InverseMatch(Matches.unitIsInTerritory(m_territory)));
    airNotInTerritory.getMatches( m_attackingUnits, Matches.UnitIsAir);

    m_attackingUnits.removeAll(Match.getMatches(m_attackingUnits,airNotInTerritory));

}
```

and the line that includes `Matches#UnitIsAir` was removed in e6bb36b with a commit message of "eclipse inspired code cleanups."

At first, I thought this was a problem because I expected `Matches#UnitIsAir` to be involved in the composite match somehow.  However, after a double-take, I realized that the removed line wasn't being added to the composite match, but was a no-op and thus had no effect on the `m_attackingUnits.removeAll()` code below.

TL;DR: This code has been in the game since 2004.  It's either in the running for longest-lived bug ever, or the method was simply misnamed when it was introduced.

Comment (ron-murhammer) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: Probably option 2. People tend to report the bugs but not the poorly named methods :)

**Comment by ron-murhammer** [comment]
@ssoloff Needs rebased.

**Comment by ssoloff** [comment]
@ron-murhammer Rebased against `master` at a87410d (1.9.0.0.5085).
